snapshot-replicator/main.tf

```diff
@@ -242,49 +235,168 @@ resource "aws_db_event_subscription" "default" {
-resource "aws_cloudwatch_event_rule" "every_24_hours" {
-  count = var.enable ? 1 : 0
-  name = "every-24-hours-${var.environment}"
-  description = "Fires every 24 hours"
+resource "aws_cloudwatch_event_rule" "invoke_rds_cleanup_lambda" {
+  count               = var.enable ? 1 : 0
+  name                = "invoke-rds-cleanup-lambda-${var.environment}"
+  description         = "Fires every 24 hours"
   schedule_expression = "rate(24 hours)"
 }
 
 resource "aws_cloudwatch_event_target" "activate_lambda_cron" {
-  count = var.enable ? 1 : 0
-  rule = aws_cloudwatch_event_rule.every_6_hours[0].name
+  count     = var.enable ? 1 : 0
+  rule      = aws_cloudwatch_event_rule.invoke_rds_snapshot_lambda[0].name
   target_id = "rds-backup"
-  arn = aws_lambda_function.rds_create_snapshot[0].arn
+  arn       = aws_lambda_function.rds_snapshot_create[0].arn
 }
 
 resource "aws_cloudwatch_event_target" "activate_lambda_removal_cron" {
-  count = var.enable ? 1 : 0
-  rule = aws_cloudwatch_event_rule.every_24_hours[0].name
+  count     = var.enable ? 1 : 0
+  rule      = aws_cloudwatch_event_rule.invoke_rds_cleanup_lambda[0].name
   target_id = "rds-retention-removal"
-  arn = aws_lambda_function.remove_snapshot_retention[0].arn
+  arn       = aws_lambda_function.rds_snapshot_cleanup[0].arn
+}
+
+
+resource "aws_lambda_permission" "cloudwatch_invoke_rds_snapshot_lambda" {
+  count         = var.enable ? 1 : 0
+  statement_id  = "AllowExecutionFromCloudWatch"
+  action        = "lambda:InvokeFunction"
+  function_name = aws_lambda_function.rds_snapshot_cleanup[0].function_name
+  principal     = "events.amazonaws.com"
+  source_arn    = aws_cloudwatch_event_rule.invoke_rds_snapshot_lambda[0].arn
+}
+
+resource "aws_lambda_permission" "cloudwatch_invoke_rds_cleanup_lambda" {
+  count         = var.enable ? 1 : 0
+  statement_id  = "AllowExecutionFromCloudWatch"
+  action        = "lambda:InvokeFunction"
+  function_name = aws_lambda_function.rds_snapshot_cleanup[0].function_name
+  principal     = "events.amazonaws.com"
+  source_arn    = aws_cloudwatch_event_rule.invoke_rds_cleanup_lambda[0].arn
+}
+
+
+
+# Cleanup of backups on remote region
+resource "aws_lambda_function" "rds_snapshot_cleanup_remote" {
+  count         = var.enable ? 1 : 0
+  provider      = aws.replica
+  function_name = "remove-snapshot-retention-${var.environment}"
+  role          = aws_iam_role.iam_for_lambda[0].arn
+  handler       = "remove_snapshots.lambda_handler"
+
+  filename         = "${path.module}/shipper.zip"
+  source_code_hash = filebase64sha256("${path.module}/shipper.zip")
+
+  runtime = "python2.7"
+  timeout = "120"
+
+  environment {
+    variables = {
+      TARGET_REGION = var.aws_destination_region
+      DB_INSTANCES  = join(",", var.db_instances)
+      RETENTION     = var.retention
+    }
+  }
 }
+
+resource "aws_iam_role" "rds_cleanup_lambda_remote" {
+  count    = var.enable ? 1 : 0
+  provider = aws.replica
+  name     = "default-lambda-${var.environment}"
```

Comment:
I would set a more meaningful name here, and the same for the existing role in the source region. Not sure why it's called `default-lambda` tbh 😅

Reply:
true :D copied it over from the first one and didn't notice